overlord/servicestate/quota_control.go

```diff
@@ -311,69 +313,16 @@ func EnsureSnapAbsentFromQuota(st *state.State, snap string) error {
 	return nil
 }
 
-// QuotaChangeConflictError represents an error because of quota group conflicts between changes.
-type QuotaChangeConflictError struct {
-	Quota      string
-	ChangeKind string
-	// a Message is optional, otherwise one is composed from the other information
-	Message string
-}
-
-func (e *QuotaChangeConflictError) Error() string {
-	if e.Message != "" {
-		return e.Message
-	}
-	if e.ChangeKind != "" {
-		return fmt.Sprintf("quota group %q has %q change in progress", e.Quota, e.ChangeKind)
-	}
-	return fmt.Sprintf("quota group %q has changes in progress", e.Quota)
-}
-
-// CheckQuotaChangeConflictMany ensures that for the given quota groups no other
-// changes that alters them (like create, update, remove) are in
-// progress. If a conflict is detected an error is returned.
-func CheckQuotaChangeConflictMany(st *state.State, quotaNames []string) error {
-	quotaMap := make(map[string]bool, len(quotaNames))
-	for _, k := range quotaNames {
-		quotaMap[k] = true
-	}
-
-	for _, task := range st.Tasks() {
-		chg := task.Change()
-		if chg == nil || chg.IsReady() {
-			continue
-		}
-
-		quotas, err := affectedQuotas(task)
-		if err != nil {
-			return err
-		}
-
-		for _, quota := range quotas {
-			if quotaMap[quota] {
-				return &QuotaChangeConflictError{Quota: quota, ChangeKind: chg.Kind()}
-			}
-		}
-	}
-
-	return nil
-}
-
-func affectedQuotas(task *state.Task) ([]string, error) {
-	// so far only quota-control is relevant
-	if task.Kind() != "quota-control" {
-		return nil, nil
+func AddSnapToQuotaGroup(st *state.State, snapName string, quotaGroup string) (*state.Task, error) {
+	if err := CheckQuotaChangeConflictMany(st, []string{quotaGroup}); err != nil {
```

Comment:
the doc comment should explain that this doesn't check in itself for snap-level conflict, because it is meant to be used in a larger context that already did that

Reply:
Done!